apps/prairielearn/python/prairielearn/core.py

```diff
@@ -1058,7 +1060,7 @@ def string_from_number_sigfig(a: _NumPyScalarType, digits: int = 2) -> str:
 
     # `np.iscomplexobj` isn't a proper type guard, so we need to use
     # casting to call this function.
-    if np.iscomplexobj(a):
+    if np.iscomplexobj(cast(Any, a)):
```

Comment:
Nit: `not isinstance(a, numbers.Number) and np.iscomplexobj(a)`

Reply:
Why? This would reintroduce the bug that I fixed.

```py
>>> import numbers
>>> isinstance(complex(1, 2), numbers.Number)
True
```

Reply:
Ah, I was not aware that those types overlapped. Sorry about that.